src/gval/utils/loading_datasets.py

```diff
@@ -439,6 +492,125 @@ def get_stac_data(
         return stack
 
 
+def _stac_to_df(stac_items: list, assets: list = None) -> pd.DataFrame:
+    """Convert a list of stac items to a DataFrame
+
+    Parameters
+    ----------
+    stac_items: list
+        List of stac items to create a catalog with
+    assets : list, default = None
+        Assets to keep, (keep all if None)
+
+    Returns
+    -------
+    pd.DataFrame
+        A DataFrame with rows for each unique item/asset combination
+
+    Raises
+    ------
+    ValueError
+        No entries in DataFrame due to nonexistent asset
+
+    """
+
+    dfs, compare_idx = [], 1
+    for stac_item in stac_items:
+        map_name, map_id, compare_id = [], [], []
+        for key, item in stac_item.assets.items():
+            if assets is None or key in assets:
+                map_name.append(key)
+                map_id.append(item.href)
+                compare_id.append(compare_idx)
+                compare_idx += 1
+
+        len_assets = len(map_name)
+
+        df_contents = {
+            "collection_id": [stac_item.collection_id] * len_assets,
+            "item_id": [stac_item.id] * len_assets,
+            "item_time": [stac_item.get_datetime()] * len_assets,
+            "create_time": [stac_item.properties["created"]] * len_assets,
+            "map_id": map_id,
+            "map_name": map_name,
+            "compare_id": compare_id,
+            "coverage_geometry_type": [stac_item.geometry["type"]] * len_assets,
+            "coverage_geometry_coords": [stac_item.geometry["coordinates"]]
+            * len_assets,
+            "coverage_epsg": ["4326"] * len_assets,
+            "asset_epsg": [stac_item.properties["proj:epsg"]] * len_assets,
+        }
+
+        dfs.append(pd.DataFrame(df_contents))
```

Comment:
Appending is a big time suck if this starts getting big. The array's have to get readdressed in memory. Anyway a df can be pre-initialized with the column names then Nones. Every row just writes the row out?